src/js/components/SpacePicker.js

```diff
@@ -34,9 +34,9 @@ export default function SpacePicker() {
     dispatch(refreshSpaceNames())
   }
 
-  function onSpaceChange(val) {
-    setSpace(val)
-    setTimeout(() => dispatch(initSpace(val)))
+  function onSpaceChange(id, name) {
+    setSpace(name)
```

Comment:
`setSpace(name)` is like an optimistic update to the UI before that space actually gets initialized below. Before it was setting that to the id. The result was a brief display of the id, then back to the name once the space was fully initialized.